src/WebJobs.Script/Description/Node/functions/functions.js

```diff
@@ -65,7 +69,12 @@ function createFunction(f) {
         inputs.unshift(context);
         delete context._inputs;
 
-        f.apply(null, inputs);
+        var userReturn = f.apply(null, inputs);
+        if (userReturn && typeof userReturn.then === 'function') {
+            context._promise = true;
+            userReturn.then((result) => context.done(null, result))
```

Comment:
Our `context.done` programming model allows a result to be returned, which is an object whose properties map to binding names (e.g. `context.done(null, { output: "FooBar"}`). It appears you're allowing a promise to conform to this and also return a result object, but I want to be sure :)